subsys/logging/Kconfig.backends

```diff
@@ -33,7 +33,7 @@ config LOG_BACKEND_UART_OUTPUT_TEXT
 	help
 	  Output in text.
 
-config LOG_BACKEND_UART_SYST_ENABLE
+config LOG_BACKEND_UART_OUTPUT_SYST
```

Comment:
@MaureenHelm Since I have defined LOG_BACKEND_UART_OUTPUT_DEFAULT which gives us the option to select from LOG_BACKEND_UART_OUTPUT_TEXT, LOG_BACKEND_UART_OUTPUT_SYST or LOG_BACKEND_UART_OUTPUT_DICTIONARY. Just wanted to be consistent with the naming of formats for backends here. I have a plan to do so for every backend. Also, this wont go away with v1 logging because from my understanding we still need the functionality that each backend can have a different logging format when multiple backends are active. Let me know if this still doesn't make sense.

Reply:
We can simplify this by having one default output Kconfig for all backends:

```
choice LOG_BACKEND_OUTPUT_DEFAULT
        prompt "Default output format for all backends"

config LOG_BACKEND_OUTPUT_DEFAULT_TEXT
        bool "Text"

config LOG_BACKEND_OUTPUT_DEFAULT_SYST
        bool "Sys-T"

config LOG_BACKEND_OUTPUT_DEFAULT_DICT
        bool "Dictionary"

endchoice
```

Or, if we need to specify a different default for each backend, then use a Kconfig template to generate them instead of hand-coding all the possible combinations. See for example:
- `subsys/logging/Kconfig.template.log_config`
- `subsys/net/Kconfig.template.log_config.net`
- `subsys/usb/class/Kconfig.template.composite_device_number`

Reply:
@MaureenHelm That was a very good suggestion. I have leveraged the Kconfig template to add configurations and added support for all the backends. Please re-review.